### ISSUE TITLE
Test failure on Rust v1.81

### ISSUE THREAD

**Issue by eserilev** [sigp/lighthouse#6405]
## Description

The test `prune_excess_peers` fails when running against Rust 1.81

```
 panicked at /Users/eitan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/quickcheck-1.0.3/src/tester.rs:165:28:

Error: "user-provided comparison function does not correctly implement a total order"

```

I believe its related to the new sort impls in 1.81 that panic if they're unable to meaningfully sort a result (instead of just returning randomly sorted data) https://blog.rust-lang.org/2024/09/05/Rust-1.81.0.html#new-sort-implementations

cc @AgeManning @jxs

**Comment by michaelsproul** [comment]
I think we need to prevent `NaN` values from creeping in as scores here:

https://github.com/sigp/lighthouse/blob/2e0eb6d1b8705bbda2ba56eb195d9cc7c6575e95/beacon_node/lighthouse_network/src/peer_manager/mod.rs#L2266-L2268

Our `PartialOrd` and `Ord` implementations for `Score` are broken for `NaN`:

https://github.com/sigp/lighthouse/blob/2e0eb6d1b8705bbda2ba56eb195d9cc7c6575e95/beacon_node/lighthouse_network/src/peer_manager/peerdb/score.rs#L340-L342

It returns `Some(Equal)` for incomparable values (any `NaN`s), which violates the first rule of [`PartialOrd`](https://doc.rust-lang.org/stable/std/cmp/trait.PartialOrd.html):

> 1. `a == b` if and only if `partial_cmp(a, b) == Some(Equal)`

We have:

```
partial_cmp(NaN, x) == Some(Equal)
```

which would require `NaN == x` (false).

**Comment by eserilev** [comment]
Closed by #6407
